src/constructs/ecs/ecs-task.ts

```diff
@@ -120,6 +120,13 @@ export interface GuEcsTaskProps extends AppIdentity {
   securityGroups?: ISecurityGroup[];
   customTaskPolicies?: PolicyStatement[];
   environmentOverrides?: TaskEnvironmentVariable[];
+  /**
+   * If your container needs to write to disk whilst running, you will need to mount a non-root volume to use. Setting
+   * volumeMountPath will ensure a volume is mounted at that location, making use of Faragate ephemeral storage (set by
+   * 'storage' param)
+   *
+   */
+  volumeMountPath?: string;
```

Comment:
I wonder if a cleaner API might be to expose the created `taskDefinition` and `containerDefinition` as properties from `GuEcsTask`:

```ts
export class GuEcsTask extends Construct {
  public readonly task: TaskDefinition; // Been lazy and made up the type here!
  public readonly container: ContainerDefinition;

  constructor() {
    super();
    const taskDefinition = new TaskDefinition(...);
    const containerDefinition = new ContainerDefinition(...);

    this.task = taskDefinition;
    this.container = containerDefinition;
  }
}
```

This approach avoids adding a new optional prop and increases flexibility as we can now call any function available on `TaskDefinition` and `ContainerDefinition`:

```ts
const { task, container } = new GuEcsTask(...);

task.addVolume({
  name: 'asd'
});

task.someOtherUsefulFunction(...);
```

Reply:
Oh that's a much better approach!